src/main.rs

```diff
@@ -47,55 +51,83 @@ fn compile(txt: &str) -> (String, Vec<usize>) {
     (out, map)
 }
 
-fn interp(prg: &[u8], map: Vec<usize>) {
-    let mut pc = 0;
-    let mut ptr = 0;
-    let mut cells = Vec::with_capacity(30000);
-    cells.resize(30000, 0u8);
-    let mut stdin = std::io::stdin().bytes();
-    let mut stdout = std::io::stdout();
-    while pc < prg.len() {
-        match prg[pc] as char {
-            '>' => {
-                if ptr == cells.len() {
-                    cells.push(0);
-                }
-                ptr += 1;
-            }
-            '<' => {
-                if ptr > 0 {
-                    ptr -= 1;
-                }
-            }
-            '+' => {
-                cells[ptr] = cells[ptr].wrapping_add(1);
-            }
-            '-' => {
-                cells[ptr] = cells[ptr].wrapping_sub(1);
+#[derive(Debug)]
+struct InterpCx {
```

Comment:
I *think* we agreed on `InterpCtx` -- at least, that's what the branch I'm working on is calling it :p